morf-core/src/main/java/org/alfasoftware/morf/sql/PostgreSQLCustomHint.java

```diff
@@ -0,0 +1,23 @@
+package org.alfasoftware.morf.sql;
+
+/**
+ * Represents a custom hint for a query
+ */
+public final class PostgreSQLCustomHint implements CustomHint {
```

Comment:
The precedence unfortunatelly has been done, by OracleCustomHint.

I do agree though that this could be changed in the future.
I am not sure in what way, but we do have PortableSqlStatement as another precedence.

Reply:
Looking at `PortableSqlStatement` we could implement the following usage pattern.

`select.writeCustomHint(PostgreSQL.IDENTIFIER, "custom logic")`

And in the dialect have:
```
      if (hint instanceof CustomHint && PostgreSQL.IDENTIFIER.equals(((CustomHint)hint).getDatabaseType()))) {
        builder.append(" ")
        .append(((CustomHint)hint).getCustomHint());
      }

```

Reply:
We could also deprecate `OracleHint` and the `SelectStatement.withCustomHint(CustomHint)` method.